src/library/dao/autodjcratesdao.cpp

```diff
@@ -881,3 +884,95 @@ void AutoDJCratesDAO::slotPlayerInfoTrackUnloaded(QString group,
         }
     }
 }
+//
+// We are selecting the track in the following manner.
+// We divide the library tracks into three section . For which
```

Comment:
There's a lot of weird spacing in this comment, please replace with:

// We are selecting the track in the following manner:
// We divide the library tracks into three sections, for which
// we sort the library according to times_played and select a
// percentage_of_prefered_tracks (70% by default ignoring the least-played 
// 15% and most played 15%). We select a random track from this 70%
// or a random track from the remaining 30%.
//
// Example : For 100 random additions this will yield:
//           6 Tracks from first 15% (not favored)
//           88 from the middle 70%  (favored)
//           6  from the last 15%    (not favored)
// There for 88 favored tracks in 100 random addition.
// This is better than returning the least played track.
// Furthermore this also does not restrict our function to only retrieve
// unplayed tracks (there is probably a reason they are unplayed).